model/shared/src/main/scala/explore/model/Asterism.scala

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.model
+
+import cats.Eq
+import cats.data.NonEmptyList
+import cats.syntax.all._
+import lucuma.core.model.Target
+import monocle._
+
+final case class Asterism(targets: NonEmptyList[TargetWithId]) {
+  def toSidereal: List[SiderealTargetWithId] =
+    targets.traverse(_.toSidereal).foldMap(_.toList)
+
+  // This should be calculatedd from the other targets or manually overriden
+  def baseTarget: TargetWithId = targets.head
+
+  def hasId(id: Target.Id): Boolean = targets.exists(_.id === id)
+}
+
+object Asterism {
+  val isoTargets: Iso[NonEmptyList[TargetWithId], Asterism] =
+    Iso[Asterism, NonEmptyList[TargetWithId]](_.targets)(Asterism.apply).reverse
+
+  val targets: Lens[Asterism, NonEmptyList[TargetWithId]] = isoTargets.reverse.asLens
+
+  val fromTargets: Lens[NonEmptyList[TargetWithId], Asterism] = isoTargets.asLens
+
+  val targetsEach: Traversal[Asterism, TargetWithId] = targets.each
+
+  val fromTargetsList: Iso[List[TargetWithId], Option[Asterism]] =
+    Iso[List[TargetWithId], Option[Asterism]](fromTargets) {
+      case Some(Asterism(targets)) => targets.toList
+      case _                       => Nil
+    }
+
+  def fromTargets(targets: List[TargetWithId]): Option[Asterism] =
+    NonEmptyList.fromList(targets).map(Asterism.apply)
+
+  def of(targets: TargetWithId): Asterism =
+    Asterism(NonEmptyList.of(targets))
+
+  def targetOptional(targetId: Target.Id): Optional[Option[Asterism], Target] =
```

Comment:
Is this lawful?

Reply:
That's a good question, I moved this here because it was duplicated in two places across the code. It would be great to test it but it takes a param. Maybe I can put the checkAll inside a forAll

Reply:
Actually you can do the test and seem to work, it is a bitslow but that's fine